src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/CannotAddOrRemoveVirtualModifier.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+#nullable enable
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the 'virtual' modifier is not added to
+    /// or removed from a member.
+    /// </summary>
+    public class CannotAddOrRemoveVirtualModifier : Rule
+    {
+        public override void Initialize(RuleRunnerContext context) => context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+
+        private void RunOnMemberSymbol(ISymbol left, ISymbol right, ITypeSymbol leftContainingType, ITypeSymbol rightContainingType, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            // Members must exist
+            if (left is null || right is null)
+            {
+                return;
+            }
+            // If the left member is virtual, it must not be sealed or inside a sealed type.
+            if (left.IsVirtual)
+            {
+                // If the right member is not virtual, emit a diagnostic
+                // that the virtual modifier cannot be removed.
+                if (!right.IsVirtual)
+                {
+                    differences.Add(new CompatDifference(
+                    DiagnosticIds.CannotRemoveVirtualModifier, string.Format(
+                        Resources.CannotRemoveVirtualModifier, left), DifferenceType.Removed, right));
+                }
+            }
```

Comment:
```suggestion
            if (left.IsVirtual && !right.IsVirtual)
            {
                // If the right member is not virtual, emit a diagnostic
                // that the virtual modifier cannot be removed.
                differences.Add(new CompatDifference(
                    DiagnosticIds.CannotRemoveVirtualModifier, string.Format(
                    Resources.CannotRemoveVirtualModifier, left), DifferenceType.Removed, right));
            }
```
NIT: no need to have separate ifchecks

Reply:
In this case, the `else if` branch assumes that `left.IsVirtual` is `false`, so combining these two conditions would break the `else`. I *could* do
```C#
if (left.IsVirtual && !right.IsVirtual) {
...
}

if (!left.IsVirtual && (Settings.StrictMode || !left.IsAbstract)) {
...
}
```
but the nested if statements meant I only had to check `left.IsVirtual` once.

Reply:
oh, I missed that.